### PR TITLE
chore(opentelemetry source): Refactor source into multiple files

### PR DESCRIPTION
**Pull request by spencergilbert**
- chore(opentelemetry source): Move tests into separate module
- chore(opentelemetry source): split config from log service

### CODE REVIEW

File: src/sources/opentelemetry/tests.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+#[cfg(test)]
+mod tests {
```
Comment (jszwedko) [inline src/sources/opentelemetry/tests.rs:2]: I think you can just remove the `mod tests { }` and move the cfg attr up into `mod.rs`.

**Review by jszwedko — APPROVED** [review]
👍 refactoring looks good to me.

File: src/sources/opentelemetry/tests.rs (hunk 1 above)
Comment (spencergilbert) [reply src/sources/opentelemetry/tests.rs:2]: Ah - missed in the copy-pasting

Comment (spencergilbert) [reply src/sources/opentelemetry/tests.rs]: addressed in [56a888a](https://github.com/vectordotdev/vector/pull/13658/commits/56a888ab1d44498d9f8b7cfce3604d443c3dc6d3)

**Review by neuronull — APPROVED** [review]

**Pull request merged**